tests/unittests/test_auth_in_discovery.py

```diff
@@ -0,0 +1,56 @@
+from unittest import mock
+from tap_stripe import stripe, Context
+import tap_stripe
+import unittest
+import requests
+import json
+
+# Mock args
+class Args():
+    def __init__(self):
+        self.discover = True
+        self.catalog = False
+        self.config = {"client_secret": "test_secret", "account_id": "test_account", "start_date": "test_start_date"}
+        self.state = False
+
+# Mock response
+def get_mock_http_response(status_code, content={}):
+    contents = json.dumps(content)
+    response = requests.Response()
+    response.status_code = status_code
+    response.headers = {}
+    response._content = contents.encode()
+    return response
+
+@mock.patch('tap_stripe.utils.parse_args')
+class TestBasicAuthInDiscoverMode(unittest.TestCase):
+    @mock.patch('tap_stripe.discover')
+    @mock.patch('stripe.http_client.requests.Session.request')
+    def test_basic_auth_no_access_401(self, mock_request, mocked_discover, mocked_args):
+        '''
+            Verify exception is raised for no access(401) error code for basic auth
+            and discover is called once for setup Context.
```

Comment:
Update comment here as discover function is not called for this test.

Reply:
Updated